subprojects/plugins/src/main/java/org/gradle/api/plugins/WarPlugin.java

```diff
@@ -63,8 +64,8 @@ public WarPlugin(ObjectFactory objectFactory, ImmutableAttributesFactory attribu
     @Override
     public void apply(final Project project) {
         project.getPluginManager().apply(JavaPlugin.class);
-        final WarPluginConvention pluginConvention = new DefaultWarPluginConvention(project);
-        project.getConvention().getPlugins().put("war", pluginConvention);
+        final DefaultWarPluginConvention pluginConvention = new DefaultWarPluginConvention(project);
+        project.getConvention().getPlugins().put("war", new NaggingWarPluginConvention(pluginConvention));
```

Comment:
Why did you take a decorator approach?
There is a chance that this breaks user code if they were casting it to `DefaultWarPluginConvention`. I know they should not but ... 🤷

Reply:
Our internal code still depends on the values defined in the convention. For example here: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/WarPlugin.java#L70

If I added the nagging directly to `DefaultWarPluginConvention` Gradle would always emit the deprecation warning when the `war` plugin is applied.

Reply:
Besides, `DefaultWarPluginConvention` does not expose additional features compared to what's available on `WarPluginConvention`. I usually use grep.app to find usages of Gradle APIs and t[here are no hits](https://grep.app/search?q=DefaultWarPluginConvention) for `DefaultWarPluginConvention`.

Reply:
Ah right, but we have `DeprecationLogger.whileDisabled` for this as well.

Reply:
I've searched for _all_ default implementation of core conventions and found that only the `DefaultJavaPluginConvention` was used in external projects ([according to grep.app](https://grep.app/search?q=DefaultJavaPluginConvention)). The palantir project should be safe as the call is done only for Gradle versions <7.0. I've [opened an issue](https://github.com/kotest/kotest-gradle-plugin/issues/42) for the other usage.